### PR TITLE
Riemann sink fixes

### PR DESCRIPTION
**Pull request by jsoriano**
Some fixes for riemann sink:
- Workarounds issue with int64 reflection in goryman
- Reuses code for metrics appending
- Avoid setting client to nil while sending events as it causes fatal errors

### CODE REVIEW

**Comment by jsoriano** [comment]
I signed it!

File: metrics/sinks/riemann/driver.go [hunk 1]
```diff
@@ -188,22 +188,26 @@ func (sink *riemannSink) sendData(dataEvents []riemann_api.Event) {
...
 		}
 	}
 	end := time.Now()
-	glog.V(4).Info("Exported %d data to riemann in %s", len(dataEvents), end.Sub(start))
+	if errors > 0 {
+		glog.V(2).Info("There were errors sending events to Riemman, forcing reconnection")
+		sink.client.Close()
+		sink.client = nil
+	}
+	glog.V(4).Infof("Exported %d events to riemann in %s", len(dataEvents)-errors, end.Sub(start))
```
Comment (piosz) [inline metrics/sinks/riemann/driver.go]: Shouldn't be here %v instead of %s?

**Comment by piosz** [comment]
LGTM, just a nit.
Thank you for the contribution!

**Comment by jsoriano** [comment]
@piosz `time` type implements `String()`, so I think it's fine with `%s`, this is how it appears in logs:
```
Exported 4001 events to riemann in 162.355751ms
```

**Comment by jsoriano** [comment]
Thanks! :)

**Pull request merged**